python/activator/middleware_interface.py

```diff
@@ -1636,45 +1694,47 @@ def _export_subset(self, exposure_ids: set[int],
             except lsst.daf.butler.registry.DataIdError as e:
                 raise ValueError("Invalid visit or exposures.") from e
 
-        with lsst.utils.timer.time_this(_log, msg="export_outputs (transfer)", level=logging.DEBUG):
             # Transfer dimensions created by ingest in case it was never done in
             # central repo (which is normal for dev).
             # Transferring governor dimensions in parallel can cause deadlocks in
             # central registry. We need to transfer our exposure/visit dimensions,
             # so handle those manually.
-            self._export_exposure_dimensions(
+            dimension_records = self._get_dimension_records_to_export(
                 self.butler,
-                self.write_central_butler,
                 where="exposure in (exposure_ids)",
                 bind={"exposure_ids": exposure_ids},
                 instrument=self.instrument.getName(),
                 detector=self.visit.detector,
             )
-            transferred = self.write_central_butler.transfer_from(
-                self.butler, datasets, transfer="copy", transfer_dimensions=False)
+
+        with lsst.utils.timer.time_this(_log, msg="export_outputs (transfer)", level=logging.DEBUG):
+            transferred = self._butler_writer.transfer_outputs(self.butler, dimension_records, list(datasets))
             _check_transfer_completion(datasets, transferred, "Uploaded")
 
         return transferred
 
     @staticmethod
-    def _export_exposure_dimensions(src_butler, dest_butler, **kwargs):
-        """Transfer dimensions generated from an exposure to the central repo.
+    def _get_dimension_records_to_export(src_butler: Butler, **kwargs) -> GroupedDimensionRecords:
+        """Retrieve dimension records generated from an exposure that need to
+        be transferred to the central repo.
 
-        In many cases the exposure records will already exist in the central
-        repo, but this is not guaranteed (especially in dev environments).
-        Visit records never exist in the central repo and are the sole
-        responsibility of Prompt Processing.
+        In many cases the exposure records retrieved here will already exist in
+        the central repo, but this is not guaranteed (especially in dev
+        environments).
 
         Parameters
         ----------
         src_butler : `lsst.daf.butler.Butler`
-            The butler from which to transfer dimension records.
-        dest_butler : `lsst.daf.butler.Butler`
-            The butler to which to transfer records.
+            The butler from which to retrieve dimension records.
```

Comment:
The name `src_butler` still seems inappropriate in the new context. For a query operation, can this just be `butler`?